mne/time_frequency/spectrum.py

```diff
@@ -891,6 +886,43 @@ def plot_topomap(
             show=show,
         )
 
+    def _prepare_data_for_plot(self, data):
+        # handle unaggregated Welch
+        if "segment" in self._dims:
+            seg_axis = self._dims.index("segment")
+            logger.info("Aggregating Welch estimates (median) before plotting...")
+            data = np.nanmedian(data, axis=seg_axis)
+        # handle unaggregated multitaper (also handles complex -> power)
+        elif "taper" in self._dims:
+            logger.info("Aggregating multitaper estimates before plotting...")
+            data = _psd_from_mt(data, self.mt_weights)
+
+        # handle complex data (should only be Welch remaining)
+        if np.iscomplexobj(data):
+            data = (data * data.conj()).real  # Scaling may be slightly off
+
+        # handle epochs
+        if "epoch" in self._dims:
+            # XXX TODO FIXME decide how to properly aggregate across repeated
+            # measures (epochs) and non-repeated but correlated measures
+            # (channels) when calculating stddev or a CI. For across-channel
+            # aggregation, doi:10.1007/s10162-012-0321-8 used hotellings T**2
+            # with a correction factor that estimated data rank using monte
+            # carlo simulations; seems like we could use our own data rank
+            # estimation methods to similar effect. Their exact approach used
+            # complex spectra though, here we've already converted to power;
+            # not sure if that makes an important difference? Anyway that
+            # aggregation would need to happen in the _plot_psd function
+            # though, not here... for now we just average like we always did.
+
+            # only log message if averaging will actually have an effect
+            if data.shape[0] > 1:
+                logger.info("Averaging across epochs before plotting...")
+            # epoch axis should always be the first axis
+            data = data.mean(axis=0)
+
+        return data
```

Comment:
For the plotting, this is handling things similar to the `_prepare_data_for_plot()` method of the `TFR` classes:
- aggregates segments (Welch)
- aggregates tapers (multitaper)
- converts complex coeffs to power
- averages over epochs

Before this was being handled separately in each plot method (`plot`, `plot_topo`, and `plot_topomap`), but with having to handle complex data I think it helps to refactor into a shared method.